### PR TITLE
feat: Improve export-terraform function to handle additional resources

### PR DESCRIPTION
**Pull request by morgante**
This makes a number of improvements to the export-terraform function for KCC resources:

1. Adds support for exporting projects and IAM bindings.
2. Adds attribution via a provider_meta block.
3. Generates variables for the org_id.
4. Includes a README in the generated output.

### CODE REVIEW

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 1]
```diff
@@ -17,43 +17,263 @@ package terraformgenerator
...
+		sdk.Logf("Failed to retrieve resource reference for %s/%s from %s", kind, name, resource.Name)
+	}
+	ref.Children = append(ref.Children, resource)
+
+	return ref
+}
+
+func (resource *terraformResource) getParentRef(path ...string) (string, string, error) {
+	paths := [][]string{
+		{"Folder", "spec", "folderRef", "name"},
```
Comment (mengqiy) [inline functions/go/export-terraform/terraformgenerator/resources.go:194]: Same comment here

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 2]
```diff
@@ -17,43 +17,263 @@ package terraformgenerator
...
+	}
+
+	for _, path := range paths {
+		name := resource.GetStringFromObject(path[1:]...)
+		if name == "" {
+			continue
+		}
+
+		kind := path[0]
+		if kind == "detect" {
```
Comment (mengqiy) [inline functions/go/export-terraform/terraformgenerator/resources.go:209]: Is `detect` a real kind? Or it's something we defined in the function? If the latter, we should define a const for it.

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 3]
```diff
@@ -17,43 +17,263 @@ package terraformgenerator
...
-	if ref.Kind == "Organization" {
+func (ref *terraformResource) GetTerraformId(prefix ...bool) string {
+	if ref.ShouldCreate() {
+		return fmt.Sprintf("google_folder.%s.name", ref.GetResourceName())
+	}
+	hasVariable := ref.variable != nil
+	usePrefix := !(len(prefix) > 0 && !prefix[0])
+	isOrg := ref.Kind == "Organization"
+
+	switch true {
```
Comment (mengqiy) [inline functions/go/export-terraform/terraformgenerator/resources.go:265]: No need to do `switch true`. Instead you can do:
```
switch {
	case !usePrefix && hasVariable:
		...
```

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 4]
```diff
@@ -17,43 +17,263 @@ package terraformgenerator
...
+	case !usePrefix && hasVariable:
+		return fmt.Sprintf("var.%s", ref.variable.Name)
+	case !usePrefix && !hasVariable:
+		return fmt.Sprintf("\"%s\"", ref.Name)
```
Comment (mengqiy) [inline functions/go/export-terraform/terraformgenerator/resources.go:269]: We can avoid using escape character here to make it more readable. We can use 
```
'"%s"'
or
`"%s"`
```

**Review by mengqiy — COMMENTED** [review]
Please update the usage doc to reflect the new features

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 5]
```diff
@@ -17,43 +17,263 @@ package terraformgenerator
...
+}
+
+// Attach parents and other references to a resource
+func (resource *terraformResource) attachReferences() error {
+	resource.References = make(map[string]*terraformResource)
+	paths := [][]string{
+		{"BillingAccount", "spec", "billingAccountRef", "external"},
+	}
+	for _, path := range paths {
+		kind := path[0]
```
[the remark this answers is not in the document]
Comment (morgante) [reply functions/go/export-terraform/terraformgenerator/resources.go:149]: I made this into a type.

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 6]
```diff
@@ -190,24 +195,24 @@ func (resource *terraformResource) getReferencedResource(kind string, path ...st
...
 		if name == "" {
 			continue
 		}
 
-		kind := path[0]
-		if kind == "detect" {
-			kind = resource.GetStringFromObject(append(path[1:len(path)-1], "kind")...)
+		kind := path.kind
+		if len(kind) <= 1 {
+			kind = resource.GetStringFromObject(append(path.path[0:len(path.path)-1], "kind")...)
```
Comment (mengqiy) [inline functions/go/export-terraform/terraformgenerator/resources.go:215]: `path.path[0:len(path.path)-1]` is actually `path.path`.

**Review by mengqiy — APPROVED** [review]
One minor comment, LGTM otherwise

File: functions/go/export-terraform/terraformgenerator/iam.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	return len(bindings) > 0
+}
+
+// Retrieve IAM bindings for a given resource
+func (resource *terraformResource) GetIAMBindings() map[string][]*iamBinding {
+	bindings := make(map[string][]*iamBinding)
+
+	for _, child := range resource.Children {
+		switch child.Kind {
+		case "IAMPolicyMember":
```
[the remark this answers is not in the document]
Comment (morgante) [reply functions/go/export-terraform/terraformgenerator/iam.go:56]: Added additional cases for other IAM policy kinds.

File: functions/go/export-terraform/terraformgenerator/resources.go (hunk 6 above)
Comment (morgante) [reply functions/go/export-terraform/terraformgenerator/resources.go:215]: They're actually not equivalent, go slices are exclusive on the top end. This is getting everything in the slice except the last element.

I added a test to verify this.

Comment (mengqiy) [reply functions/go/export-terraform/terraformgenerator/resources.go]: You are right. I misread it.

**Review by mengqiy — APPROVED** [review]
LGTM. Will merge after CI finishes

**Pull request merged**